cmd/attest_test.go

```diff
@@ -0,0 +1,92 @@
+package cmd
+
+import (
+	"os"
+	"testing"
+
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/internal/test"
+)
+
+func makeOutputFile(tb testing.TB, output string) string {
+	tb.Helper()
+	file, err := os.CreateTemp("", output)
+	if err != nil {
+		tb.Fatal(err)
+	}
+	defer file.Close()
+	return file.Name()
+}
+
+func TestNonce(t *testing.T) {
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+
+	// Without nonce.
+	RootCmd.SetArgs([]string{"attest", "--key", "AK"})
+	if err := RootCmd.Execute(); err == nil {
+		t.Error(err)
+	}
+
+	// With odd length nonce.
+	RootCmd.SetArgs([]string{"attest", "--nonce", "12345", "--key", "AK"})
+	if err := RootCmd.Execute(); err == nil {
+		t.Error(err)
+	}
+}
+
+func TestAttestPass(t *testing.T) {
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+
+	tests := []struct {
+		name     string
+		key      string
+		algo     string
+		nonce    string
+		teenonce string
+	}{
+		{"DefaultKey", "", "rsa", "2222", ""},
+		{"Withnonce", "AK", "ecc", "2222", ""},
+	}
+
+	for _, op := range tests {
+		t.Run(op.name, func(t *testing.T) {
+			secretFile1 := makeOutputFile(t, "attest.binarypb")
+
+			attestArgs := []string{"attest", "--output", secretFile1}
+			if op.key != "" {
+				attestArgs = append(attestArgs, "--key", op.key)
+			}
+			if op.algo != "" {
+				attestArgs = append(attestArgs, "--algo", op.algo)
+			}
+			if op.nonce != "" {
+				attestArgs = append(attestArgs, "--nonce", op.nonce)
+			}
+			if op.teenonce != "" {
+				attestArgs = append(attestArgs, "--teenonce", op.teenonce)
+			}
+
+			RootCmd.SetArgs(attestArgs)
+			if err := RootCmd.Execute(); err != nil {
+				t.Error(err)
+			}
+
+		})
+	}
+}
+
+func TestAttestTeeNonceFail(t *testing.T) {
+	rwc := test.GetTPM(t)
+	defer client.CheckedClose(t, rwc)
+	ExternalTPM = rwc
+
+	// non-nil TEENonce when TEEDevice is nil
+	RootCmd.SetArgs([]string{"attest", "--nonce", "1234", "--key", "AK", "--teenonce", "12345678"})
+	if err := RootCmd.Execute(); err == nil {
+		t.Error(err)
```

Comment:
same here, error is nil in the error

Reply:
I have updated it with a message instead of returning the nil error.